ember-toucan-core/src/components/form/file-input-field.ts

```diff
@@ -0,0 +1,85 @@
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { assert } from '@ember/debug';
+import { action } from '@ember/object';
+
+// Note, unlike other inputs the first parameter is not @value but @files
+export type OnFileUploadChangeCallback = (
+  files: File[],
+  e: Event | InputEvent
+) => void;
+
+interface ToucanFormFileInputFieldComponentSignature {
+  Element: HTMLInputElement;
+  Args: {
+    accept?: string;
+    error?: string;
+    files?: File[];
+    hint?: string;
+    label: string;
+    isDisabled?: boolean;
+    multiple?: boolean;
+    onChange?: OnFileUploadChangeCallback;
+    onDeleteFile?: (file: File) => void;
+    rootTestSelector?: string;
+  };
+  Blocks: {
+    triggerText: [];
+  };
+}
+
+export default class ToucanFormFileInputFieldComponent extends Component<ToucanFormFileInputFieldComponentSignature> {
+  @tracked files: File[] = [];
```

Comment:
1. nit: you _could_ have assigned the arg here also, instead of separately in the constructor below, like 
```suggestion
  @tracked files: File[] = this.args.files ?? [];
```
2. so relating to the discussion below on "mixing args and tracked state", this is indeed a different case than `accept` and the other getters, and I wonder if that is really what we want here.
We could instead also treat this component as "controlled" in the react sense, and not have any local state. Which means 
  * we only look at the `@files` arg, even for rendering the files list 
  * `onChange` the user would be responsible to update the file list, which is state that _they own_, and let it flow back into `@files`